packages/mobile/src/components/CurrencyDisplay.tsx

```diff
@@ -2,61 +2,128 @@ import colors from '@celo/react-components/styles/colors'
+export default function CurrencyDisplay({
+  type,
+  size,
+  useColors,
+  hideSymbol,
+  amount,
+  formatAmount,
+  style,
+}: Props) {
+  const localCurrencyCode = useLocalCurrencyCode()
+  // tslint:disable-next-line: react-hooks-nesting
+  const localValue = useDollarsToLocalAmount(amount.amount) || 0
+
+  const currency =
+    amount.currencyCode === CURRENCIES[CURRENCY_ENUM.GOLD].code
+      ? CURRENCY_ENUM.GOLD
+      : CURRENCY_ENUM.DOLLAR
 
-  render() {
-    const { size, type, amount, currencySymbol: symbol } = this.props
+  // For now only show the local amount when original currency is dollars
+  const localAmount =
+    currency === CURRENCY_ENUM.DOLLAR && localCurrencyCode
+      ? amount.localAmount ?? { amount: localValue, currencyCode: localCurrencyCode }
+      : null
+  const displayAmount = localAmount ?? amount
+  const currencySymbol =
+    displayAmount === localAmount
+      ? LocalCurrencySymbol[displayAmount.currencyCode as LocalCurrencyCode]
+      : CURRENCIES[currency].symbol
+  const value = new BigNumber(displayAmount.amount)
+  const sign = value.isNegative() ? '-' : ''
+  const formattedValue = formatAmount(value.absoluteValue())
+
+  const color = useColors
+    ? currency === CURRENCY_ENUM.GOLD
+      ? colors.celoGold
+      : colors.celoGreen
+    : colors.darkSecondary
+
+  if (type === DisplayType.Big) {
+    // In this type the symbol is displayed as superscript
+    // the downside is we have to workaround React Native not supporting it
+    // and have to involve a View, which prevents this type to be embedded into a Text node
+    // see https://medium.com/@aaronmgdr/a-better-superscript-in-react-native-591b83db6caa
     const fontSize = size
-    const dollarStyle = { fontSize, lineHeight: Math.round(fontSize * 1.3), color: this.color() }
-    const currencySymbol = type === CURRENCY_ENUM.GOLD ? CURRENCIES[type].symbol : symbol
+    const symbolStyle = getBigSymbolStyle(fontSize, color)
+    const amountStyle = { fontSize, lineHeight: Math.round(fontSize * 1.3), color }
+
     return (
-      <View style={styles.container}>
-        <Text numberOfLines={1} style={[fontStyles.regular, this.symbolStyle(fontSize)]}>
-          {currencySymbol}
-        </Text>
-        <Text numberOfLines={1} style={[styles.currency, fontStyles.regular, dollarStyle]}>
-          {getMoneyDisplayValue(amount)}
+      <View style={[styles.bigContainer, style]}>
+        {!hideSymbol && (
+          <Text numberOfLines={1} style={[fontStyles.regular, symbolStyle]}>
+            {currencySymbol}
+          </Text>
+        )}
+        <Text numberOfLines={1} style={[fontStyles.regular, styles.bigCurrency, amountStyle]}>
+          {formattedValue}
         </Text>
       </View>
     )
   }
+
+  return (
+    <Text numberOfLines={1} style={[styles.container, { color }, style]}>
+      {sign}
+      {!hideSymbol && currencySymbol}
+      {formattedValue}
+    </Text>
+  )
+}
+
+CurrencyDisplay.defaultProps = {
```

Comment:
If we're setting these default props, should these values be made optional with `?` in the interface above?

Reply:
TypeScript does this automatically when consuming this component from JSX.